cirq/google/calibration/phased_fsim.py

```diff
@@ -130,3 +140,217 @@ def override_by(self, other: 'PhasedFSimCharacterization') -> 'PhasedFSimCharact
+        """Decodes the characterization result issued for this request."""
+
+
+@json_serializable_dataclass(frozen=True)
+class FloquetPhasedFSimCalibrationOptions:
+    """Options specific to Floquet PhasedFSimCalibration.
+
+    Some angles require another angle to be characterized first so result might have more angles
+    characterized than requested here.
+
+    Attributes:
+        characterize_theta: Whether to characterize θ angle.
+        characterize_zeta: Whether to characterize ζ angle.
+        characterize_chi: Whether to characterize χ angle.
+        characterize_gamma: Whether to characterize γ angle.
+        characterize_phi: Whether to characterize φ angle.
+    """
+
+    characterize_theta: bool
+    characterize_zeta: bool
+    characterize_chi: bool
+    characterize_gamma: bool
+    characterize_phi: bool
+
+
+@dataclasses.dataclass(frozen=True)
+class FloquetPhasedFSimCalibrationResult(PhasedFSimCalibrationResult):
+    """PhasedFSim characterization result specific to Floquet calibration.
+
+    Attributes:
+        options: Options of the characterization from the request.
+    """
+
+    options: FloquetPhasedFSimCalibrationOptions
+
+    @classmethod
+    def _from_json_dict_(
+        cls,
+        parameters: List[Tuple[Qid, Qid, PhasedFSimCharacterization]],
+        gate: Gate,
+        **kwargs,
+    ) -> 'PhasedFSimCalibrationResult':
+        """Magic method for the JSON serialization protocol.
+
+        Converts serialized dictionary into a dict suitable for
+        class instantiation."""
+        return cls(cls._create_parameters_dict(parameters), gate, kwargs['options'])
+
+    def _json_dict_(self) -> Dict[str, Any]:
+        """Magic method for the JSON serialization protocol."""
+        result_dict = super()._json_dict_()
+        result_dict['cirq_type'] = 'FloquetPhasedFSimCalibrationResult'
+        result_dict['options'] = self.options
+        return result_dict
+
+
+@dataclasses.dataclass(frozen=True)
+class FloquetPhasedFSimCalibrationRequest(PhasedFSimCalibrationRequest):
+    """PhasedFSim characterization request specific to Floquet calibration.
+
+    Attributes:
+        options: Floquet-specific characterization options.
+    """
+
+    options: FloquetPhasedFSimCalibrationOptions
+
+    def to_calibration_layer(self) -> CalibrationLayer:
+        circuit = Circuit([self.gate.on(*pair) for pair in self.pairs])
+        return CalibrationLayer(
+            calibration_type=_FLOQUET_PHASED_FSIM_HANDLER_NAME,
+            program=circuit,
+            args={
+                'est_theta': self.options.characterize_theta,
+                'est_zeta': self.options.characterize_zeta,
+                'est_chi': self.options.characterize_chi,
+                'est_gamma': self.options.characterize_gamma,
+                'est_phi': self.options.characterize_phi,
+                'readout_corrections': True,
```

Comment:
any reason why this isn't part of `self.options`? Can we change either the fields of the `..Options` or the API arg-names so this can be something like `args=asdict(self.options)`?

Reply:
This is experimental feature which is not ready to be played with yet.

Reply:
ok. It would be nice for future experimental features if the `args` was "forwards compatible" so one doesn't have to sneak in this option under-the-hood